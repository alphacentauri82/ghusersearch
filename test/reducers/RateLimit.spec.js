import rateLimitReducer from 'reducers/RateLimit';

describe('Reducer: rateLimit', () => {

  describe('when passed no state', () => {
    it('should return the initial state', () => {
      expect(
        rateLimitReducer(undefined, {})
      ).toMatchSnapshot();
    });
  });

  describe('when a rate limit request is successful', () => {
    it('should update the state with the response', () => {
      const beforeState = {
        core: {},
        search: {},
      };
      const action = {
        payload: {
          core: {test: 'foo'},
          search: {test: 'bar'},
        },
        type: 'API_RATE_LIMIT_SUCCESS',
      };
      const afterState = rateLimitReducer(beforeState, action);

      expect(
        afterState
      ).toMatchSnapshot();
      expect(afterState).not.toBe(beforeState);
    });
  });

  describe('when a search request fails', () => {
    it('should update the state with the error message', () => {
      const beforeState = {};
      const err = new Error('the request died');
      const action = {
        error: true,
        payload: err,
        type: 'API_RATE_LIMIT_FAILURE',
      };
      const afterState = rateLimitReducer(beforeState, action);

      expect(
        afterState
      ).toMatchSnapshot();
      expect(afterState).not.toBe(beforeState);
    });
  });

});
