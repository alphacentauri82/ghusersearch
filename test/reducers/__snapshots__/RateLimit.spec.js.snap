// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer: rateLimit when a rate limit request is successful should update the state with the response 1`] = `
Object {
  "core": Object {
    "test": "foo",
  },
  "search": Object {
    "test": "bar",
  },
}
`;

exports[`Reducer: rateLimit when a search request fails should update the state with the error message 1`] = `
Object {
  "payload": [Error: the request died],
}
`;

exports[`Reducer: rateLimit when passed no state should return the initial state 1`] = `
Object {
  "core": Object {},
  "search": Object {},
}
`;
